src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -763,6 +765,7 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
                                         kind: Box::new(
                                             PatternKind::AscribeUserType {
                                                 subpattern: pattern,
+                                                variance: ty::Variance::Contravariant,
```

Comment:
Here we should add a comment like 

"Note that use *Contravariant* here. See the `variance` field for details."

Or perhaps move that section of the `variance` field comment to here.

Reply:
Added.